common/txmgr/tracker.go

```diff
@@ -92,8 +92,7 @@ func NewTracker[
 }
 
 func (tr *Tracker[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Start(_ context.Context) (err error) {
-	tr.lock.Lock()
-	defer tr.lock.Unlock()
```

Comment:
Just want to be wary of removing the lock entirely from startup. If you're sure we don't need one then I'm ok removing it. Otherwise, maybe this would be better served in the `startInternal` following the pattern of other components like the Broadcaster and Confirmer?

Reply:
Yeah we should keep the lock, I'll move it to internal. Since `GetAbandonedAddresses()` is called externally and needs to lock we should probably ensure the startup lock remains. I don't think it's likely to be detrimental, but I'd feel better keeping it.